ambari-server/src/main/java/org/apache/ambari/server/stack/upgrade/Grouping.java

```diff
@@ -229,15 +229,8 @@ private void addTasksToStageInBatches(List<TaskWrapper> tasks, String verb, Upgr
       for (TaskWrapper tw : tasks) {
         List<Set<String>> hostSets = null;
 
-        if (m_grouping.parallelScheduler != null) {
-          int taskParallelism = m_grouping.parallelScheduler.maxDegreeOfParallelism;
-          if (taskParallelism == Integer.MAX_VALUE) {
-            taskParallelism = ctx.getDefaultMaxDegreeOfParallelism();
-          }
-          hostSets = SetUtils.split(tw.getHosts(), taskParallelism);
-        } else {
-          hostSets = SetUtils.split(tw.getHosts(), 1);
-        }
+        int parallel = getParallelHostCount(ctx, 1);
```

Comment:
Why doesn't this grouping use the `OrchestraationOptions.getConcurrencyCount()` like the `MultiHomedBuilder` does?

Reply:
Realized that it's because a grouping can have multiple services/components

Reply:
The default `Grouping` isn't really used for slaves, and it'd be weird to have the option for, say, Oozie servers (maybe 2 in the cluster) or other "specialty" services.